src/main/java/net/snowflake/client/core/SFJsonResultSet.java

```diff
@@ -250,4 +256,242 @@ public Date getDate(int columnIndex, TimeZone tz) throws SFException {
+        case Types.LONGNVARCHAR:
+          JsonStringToTypeConverter varcharConverter = value -> value.toString();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, varcharConverter).toArray(String[]::new));
+        case Types.BINARY:
+          JsonStringToTypeConverter bytesConverter =
+              value ->
+                  converters.getBytesConverter().getBytes(value, columnType, Types.BINARY, scale);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, bytesConverter).toArray(Object[]::new));
+        case Types.FLOAT:
+        case Types.REAL:
+          JsonStringToTypeConverter floatConverter =
+              value -> converters.getNumberConverter().getFloat(value, columnType);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, floatConverter).toArray(Float[]::new));
+        case Types.DOUBLE:
+          JsonStringToTypeConverter doubleConverter =
+              value -> converters.getNumberConverter().getDouble(value, Types.DOUBLE);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, doubleConverter).toArray(Double[]::new));
+        case Types.DATE:
+          JsonStringToTypeConverter dateConverter = createDateConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, dateConverter).toArray(Date[]::new));
+        case Types.TIME:
+          JsonStringToTypeConverter timeConverter = createTimeConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, timeConverter).toArray(Time[]::new));
+        case Types.TIMESTAMP:
+          JsonStringToTypeConverter timestampConverter =
+              createTimestampConverter(columnSubType, columnType, scale);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, timestampConverter).toArray(Timestamp[]::new));
+        case Types.BOOLEAN:
+          JsonStringToTypeConverter booleanConverter =
+              value -> converters.getBooleanConverter().getBoolean(value, columnType);
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, booleanConverter).toArray(Boolean[]::new));
+        case Types.STRUCT:
+          JsonStringToTypeConverter structConverter = createStructConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, structConverter).toArray(Map[]::new));
+        case Types.ARRAY:
+          JsonStringToTypeConverter arrayConverter = createArrayConverter();
+          return new SfSqlArray(
+              columnSubType, getStream(nodeElements, arrayConverter).toArray(Map[][]::new));
+        default:
+          return null;
+      }
+    } catch (JsonProcessingException e) {
+      throw new SFException(e, ErrorCode.INVALID_STRUCT_DATA);
+    }
+  }
+
+  private static JsonStringToTypeConverter createArrayConverter() {
```

Comment:
All of this converters are just statless lambdas. Maybe we can initialize them on object init? Or one step forward - maybe we can init extract them to separate class? SRP FTW!

Reply:
Converters (functions) constructions extracted to Converters.java class.